serialization/class_definition.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package serialization
+
+import "github.com/hazelcast/hazelcast-go-client/hzerrors"
+
+// ClassDefinition defines a class schema for Portable structs.
+type ClassDefinition struct {
+	Fields    map[string]FieldDefinition
+	FactoryID int32
+	ClassID   int32
+	Version   int32
+}
+
+func NewClassDefinition(factoryID int32, classID int32, version int32) *ClassDefinition {
+	return &ClassDefinition{
+		FactoryID: factoryID,
+		ClassID:   classID,
+		Version:   version,
+		Fields:    make(map[string]FieldDefinition),
+	}
+}
+
+func (cd *ClassDefinition) AddField(definition FieldDefinition) {
```

Comment:
Just to note, this API is private on Java

Reply:
Also, I feel like exposing this to the user brings a burden to him. One needs to maintain correct indexes on FieldDefinitions, put factory id, class id, version, etc. to every definition ...

Do you see any advantage of this API over the simple `AddXxxField` methods?

Reply:
It's being used in an internal package, so it should be exported. It's possible to make it an internal standalone function, but quite a few tests depend on it. I think it would be better to create a new PR when we do that.

Reply:
I am okay with doing that later, maybe we can create an issue to not forget to make this private before the stable release